### PR TITLE
Extend the TaskEngine interface with the GetTaskByArn method

### PR DESCRIPTION
**Pull request by aaithal**
r? @samuelkarp @juanrhenals

### CODE REVIEW

**Comment by samuelkarp** [comment]
```
agent/stats/engine_test.go:314: cannot use taskEngine (type *MockTaskEngine) as type engine.TaskEngine in argument to statsEngine.MustInit:

	*MockTaskEngine does not implement engine.TaskEngine (missing GetTaskByArn method)
```

Can you regenerate the mocks?

**Comment by aaithal** [comment]
Yeah, noticed that. Regenerating mocks now.

**Comment by aaithal** [comment]
Updated PR now passes tests. r? @samuelkarp @juanrhenals

**Comment by samuelkarp** [comment]
:+1:

File: agent/stats/engine_test.go [hunk 1]
```diff
@@ -54,6 +54,10 @@ func (engine *MockTaskEngine) ListTasks() ([]*api.Task, error) {
 	return nil, nil
 }
 
+func (engine *MockTaskEngine) GetTaskByArn(arn string) (*api.Task, bool) {
```
Comment (juanrhenals) [inline agent/stats/engine_test.go]: Nit: is there any particular reason why you're returning a `bool` type here instead of just using `error`?

**Comment by juanrhenals** [comment]
Just had one nit. LGTM :+1:

File: agent/stats/engine_test.go (hunk 1 above)
Comment (aaithal) [inline agent/stats/engine_test.go]: Yeah, just following the [map lookup semantics](https://blog.golang.org/go-maps-in-action) in golang.

**Pull request merged**